packages/hooks/src/useStorageState.ts

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react';
+import { isServer } from './util';
+
+/**
+ *
+ */
+export function useStorageState<T>(
+  key: string,
+  defaultValue: T,
+  storage: Storage
+): [T, (newValue: T) => void] {
+  const rawValue = storage.getItem(key);
+  const saveItem = (newValue: T) => {
+    storage.setItem(key, JSON.stringify(newValue));
+    return newValue;
+  };
+
+  const retrieved =
+    rawValue === null ? saveItem(defaultValue) : (JSON.parse(rawValue) as T);
+  const [value, setValue] = useState(retrieved);
+
+  const setAndSaveValue = (newValue: T) => {
+    saveItem(newValue);
+    setValue(newValue);
+  };
+
+  return [value, setAndSaveValue];
+}
+
+export function useSessionStorageState<T>(
+  key: string,
+  defaultValue: T
+): [T, (newValue: T) => void] {
+  if (isServer()) {
+    return [defaultValue, () => {}];
+  }
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return useStorageState(key, defaultValue, sessionStorage);
+}
+
+export function useLocalStorageState<T>(
```

Comment:
What are your thoughts on marking these all as `@deprecated`, and instead using the ones from `react-use`? I skimmed the list and it looked like they have equivalents.

```suggestion
/**
 * @deprecated Use {XYZ} from `react-use` instead.
 */
export function useLocalStorageState<T>(
```

Reply:
why would i do that if we don't use these yet?

Reply:
seems like i should just close this and we'll use `react-use` in the main app. i can make those changes in my other branch